lib/ansible/modules/system/filesystem.py

```diff
@@ -110,13 +115,32 @@ def _get_fs_size(fssize_cmd, dev, module):
                     block_size = int(col[2].split()[0])
                     block_count = int(col[3].split(',')[0])
                     break
-        else:
-            module.fail_json(msg="Failed to get block count and block size of %s with %s" % (dev, cmd), rc=rc, err=err )
     elif 'btrfs' == fssize_cmd:
         #ToDo
         # There is no way to get the blocksize and blockcount for btrfs filesystems
         block_size = 1
         block_count = 1
+    elif fssize_cmd == 'swapon':
+        # Old swapon versions support only: swapon -s
+        # which is the same than parsing /proc/swaps;
+        # ie: awk '$1 == "$DEVICE_PATH" {print $3}' /proc/swaps
+        # and in which the size is expressed in kB.
+        # Newer swapon versions should use: swapon --noheadings --show --raw --bytes
+        rc, size, err = module.run_command("%s -s" % (cmd, ))
+        if rc == 0:
+            for line in size.splitlines():
+                row = line.split()
+                if row[0] == dev:
+                    block_count = int(row[2])
+                    block_size = 1024
+                    break
+                else:
+                    # The command can only tell the swap size when the device
+                    # is in use, otherwise return default.
+                    block_count = block_size = 1
+
+        if rc:
```

Comment:
This block should probably be unindented one level to work for all filesystem types.